### DISCUSSION TITLE
Access java (as dependency) and a jar from the installation zip file

### DISCUSSION THREAD

**Discussion by devton-here** [Homebrew/discussions#3283] [Writing Formulae/Casks] [answered]
I'm trying to provide my CLI app as a bottle via brew.
It's not open source, so I'm hosting it in my own repository.

In the Formula I've added this stuff:

```
  depends_on "openjdk@8"

  def install
    bin.install "my-app"
  end
```

The URL in the Formula points to a universal zip which contains:

- a LICENSE file
- a jar file (my-app-v1.2.3.jar)
- a bash script, just called `my-app`
- a bat file for windows (my-app.bat)

Installation with `brew install` was fine:

but my app cannot be started. Of course, the bash script is not using java from brew.
That's why I see just 

> The operation couldn’t be completed. Unable to locate a Java Runtime.
> Please visit http://www.java.com for information on installing Java.

I wonder what is the best way to access my dependency (openjdk@8).
Maybe: `#{HOMEBREW_PREFIX}/Cellar/openjdk@8/1.8.0+322/bin/java`?

Is the `$HOMEBREW_PREFIX` variable available when running `my-app` actually?

Then I've realized that my `/usr/local/Cellar/my-app/11.1.0` directory does only contain the bash script but not the jar file.

Is it correct to extend the installation method like this?

```
  def install
    bin.install "my-app"
    bin.install "my-app-v1.2.3.jar"
  end
```

or is there any other similar command to move files from the zip to the Cellar?

**Comment by SMillerDev** [comment, marked as answer]
Usually formula make their own bash script for this. See usage of `write_jar_script`. 

For example:
```rb
bin.write_jar_script libexec/"jenkins.war", "jenkins", java_version: "11"
```

**Reply by devton-here** [reply]
Great, that helped!

I need also to pass some java options, I've tried to pass it via `java_opts: "-Dfile.encoding=UTF8"` but during installation I got this error;:

> Error: An exception occurred within a child process:
>  ArgumentError: unknown keyword: java_opts

Then I've changeed it to `bin.write_jar_script libexec/"cli.jar", "olp", "-Dfile.encoding=UTF8"` and it worked.

---

I have two other questions:
  
1. I have a LICENSE file which is automatically picked up by brew to the Cellar directory. But I have an additional NOTICE file which I would like to put next to the LICSENSE file in the Cellar. How is that possible?
2. I've noticed that I need to append a suffix to the class if I want to provide multiple versions. The syntax is `AT<MAJOR><MINOR>`. I wonder how this can work correctly for these cases: `1.11` and `11.1`?

**Reply by SMillerDev** [reply]
>  But I have an additional NOTICE file which I would like to put next to the LICSENSE file in the Cellar. How is that possible?

`prefix.install "file"`

> I've noticed that I need to append a suffix to the class if I want to provide multiple versions. The syntax is AT<MAJOR><MINOR>. I wonder how this can work correctly for these cases: 1.11 and 11.1?

I'm not sure what that means

**Reply by devton-here** [reply]
I will create a new  question since it's not directly related. Thank you so much for your fast help!
